src/plugins/audio-waveform/audio-waveform-generator-dialog.js

```diff
@@ -0,0 +1,219 @@
+  };
+
+  const handleCdnResourcePickerSelect = async sourceUrl => {
+    setIsCdnResourcePickerVisible(false);
+
+    const audioContext = new AudioContext();
+
+    const sourceType = getMediaSourceType({ sourceUrl, cdnRootUrl: clientConfig.cdnRootUrl });
+    const resolvedUrl = urlUtils.getMediaUrl({ sourceUrl, sourceType, cdnRootUrl: clientConfig.cdnRootUrl });
+
+    const response = await httpClient.get(resolvedUrl, {
+      responseType: 'blob',
+      withCredentials: sourceType === MEDIA_SOURCE_TYPE.internal
+    });
+
+    setIsGeneratingPeaks(true);
+    setGeneratedPeaks(await extractPeaks(response.data, audioContext, IMAGE_OPTIMIZATION_THRESHOLD_WIDTH));
+    setIsGeneratingPeaks(false);
+  };
+
+  const handleCdnResourcePickerClose = () => {
+    setIsCdnResourcePickerVisible(false);
+  };
+
+  const segmentsDropzoneClasses = classNames({
+    'AudioWaveformGeneratorDialog-dropzone': true,
+    'u-can-drop': dropzone.isDragAccept && !isGeneratingPeaks,
+    'u-cannot-drop': dropzone.isDragReject || (dropzone.isDragAccept && isGeneratingPeaks)
+  });
+
+  return (
+    <Fragment>
+      <Modal
+        centered
+        width="80%"
+        onOk={handleOk}
+        visible={visible}
+        onCancel={onCancel}
+        title={t('dialogTitle')}
+        okText={t('common:apply')}
+        className="AudioWaveformGeneratorDialog"
+        okButtonProps={{ disabled: !imageUrl || isGeneratingPeaks, loading: isUploadingFile }}
+        >
+        <Spin spinning={isGeneratingPeaks}>
+          <div {...dropzone.getRootProps({ className: segmentsDropzoneClasses })}>
+            <input {...dropzone.getInputProps()} hidden />
+            <div
+              className="AudioWaveformGeneratorDialog-previewArea"
+              style={{ backgroundImage: imageUrl ? cssUrl(imageUrl) : 'none' }}
+              >
+              {!imageUrl && !isGeneratingPeaks && t('dialogDropzoneInfo')}
+            </div>
+            {!isGeneratingPeaks && (
+            <div className="AudioWaveformGeneratorDialog-dropzoneInfo">
+              <Tooltip title={t('dialogDropzoneInfo')}>
+                <InfoCircleOutlined className="u-info-icon" />
+              </Tooltip>
+            </div>
+            )}
+          </div>
+        </Spin>
+        <div className="AudioWaveformGeneratorDialog-controls">
+          <div>
+            <span>{t('dialogLocalFilePickerPenColorLabel')}: </span>
+            <ColorPicker color={waveformPenColor} onChange={setWaveformPenColor} width={382} inline />
+          </div>
+          <div>
+            <span>{t('dialogLocalFilePickerBaselineColorLabel')}: </span>
+            <ColorPicker color={waveformBaselineColor} onChange={setWaveformBaselineColor} width={382} inline />
+          </div>
+          <div>
+            <span>{t('dialogLocalFilePickerBackgroundColorLabel')}: </span>
+            <ColorPicker color={waveformBackgroundColor} onChange={setWaveformBackgroundColor} width={382} inline />
```

Comment:
A very particular number, this 382 😄 What does it mean?

Reply:
It overrode the default of 320 in all cases :rofl: I changed it so the default is now 382 and the props are not set anywhere. (It's the width of the color swatches dropdown btw).

Reply:
> Works very nicely! clap
> 
> Just 2 tiny things I found while testing:
> 
> * the info icon overlaps a bit with the background in the dialog

I removed this tooltip/icon completely now as it was redundant (the same text is shown in the preview area initially)

> * when you select a wrong/non-audio format in the dialog from the CDN, it throws and does not stop the spinner; maybe we should display a message to the user to draw attention to having not picked an audio file and stop the loading spinner

Fixed! There's now proper error handling. Thanks!